src/main/java/graphql/schema/idl/WiringFactory.java

```diff
@@ -53,18 +54,50 @@
      * This is called to ask if this factory can provide a data fetcher for the definition
      *
      * @param registry   the registry of all types
+     * @param parentType the type of the parent
      * @param definition the field definition in play
      * @return true if the factory can give out a date fetcher
      */
-    boolean providesDataFetcher(TypeDefinitionRegistry registry, FieldDefinition definition);
+    default boolean providesDataFetcher(TypeDefinitionRegistry registry, TypeDefinition parentType, FieldDefinition definition) {
```

Comment:
@andimarek - has this gone out in API yet?  I cant remember

if it hasnt then there is no need to preserve API.  

If it has can we make this a WiringParameters type interface

   interface WiringParameters {
        TypeDefinitionRegistry registry()
        TypeDefinition parentType()
         FieldDefinition definition()
   }

That way later when we want another parameter we can extend  it onto the interface and no one gets broken!

Since this is a SPI its ever more important